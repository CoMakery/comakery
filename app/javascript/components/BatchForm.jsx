import React from 'react'
import PropTypes from 'prop-types'
import Layout from './layouts/Layout'
import {fetch as fetchPolyfill} from 'whatwg-fetch'
import InputFieldDropdown from './styleguide/InputFieldDropdown'
import InputFieldWhiteDark from './styleguide/InputFieldWhiteDark'
import InputFieldDescription from './styleguide/InputFieldDescription'
import InputFieldDescriptionMiddle from './styleguide/InputFieldDescriptionMiddle'
import InputFieldUploadFile from './styleguide/InputFieldUploadFile'
import Button from './styleguide/Button'
import ButtonBorder from './styleguide/ButtonBorder'
import Flash from './layouts/Flash'

class BatchForm extends React.Component {
  constructor(props) {
    super(props)

    this.errorAdd = this.errorAdd.bind(this)
    this.errorRemove = this.errorRemove.bind(this)
    this.disable = this.disable.bind(this)
    this.enable = this.enable.bind(this)
    this.handleFieldChange = this.handleFieldChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)

    this.state = {
      flashMessages        : [],
      errors               : {},
      disabled             : {},
      formAction           : this.props.formAction,
      formUrl              : this.props.formUrl,
      closeOnSuccess       : false,
      'batch[specialty_id]': this.props.batch.specialtyId || Object.values(this.props.specialties)[0],
      'batch[name]'        : this.props.batch.name || '',
      'batch[goal]'        : this.props.batch.goal || '',
      'batch[description]' : this.props.batch.description || ''
    }
  }

  goBack() {
    typeof window === 'undefined' ? null : window.location = document.referrer
  }

  errorAdd(n, e) {
    this.setState({
      errors: Object.assign({}, this.state.errors, {[n]: e})
    })
  }

  errorRemove(n) {
    let e = this.state.errors
    delete e[n]
    this.setState({
      errors: e
    })
  }

  disable(a) {
    let d = this.state.disabled
    a.forEach(n => d = Object.assign({}, d, {[n]: true}))
    this.setState({
      disabled: d
    })
  }

  enable(a) {
    let d = this.state.disabled
    a.forEach(n => delete d[n])
    this.setState({
      disabled: d
    })
  }

  handleFieldChange(event) {
    this.setState({ [event.target.name]: event.target.value })

    if (!event.target.checkValidity()) {
      this.errorAdd(event.target.name, 'invalid value')
      return
    } else {
      this.errorRemove(event.target.name)
    }

    if (event.target.value === '') {
      return
    }
  }

  handleSubmit(event) {
    event.preventDefault()

    this.disable(['batch[submit]', 'batch[submit_and_close]'])

    if (!event.target.checkValidity()) {
      this.enable(['batch[submit]', 'batch[submit_and_close]'])
      return
    }

    const formData = new FormData(event.target)

    fetchPolyfill(this.state.formUrl, {
      credentials: 'same-origin',
      method     : this.state.formAction,
      body       : formData,
      headers    : {
        'X-Key-Inflection': 'snake'
      }
    }).then(response => {
      if (response.status === 200) {
        if (this.state.closeOnSuccess) {
          window.location = this.props.urlOnSuccess
        } else {
          response.json().then(data => {
            if (this.state.formAction === 'POST') {
              this.setState(state => ({
                formAction   : 'PUT',
                formUrl      : `/projects/${this.props.projectId}/batches/${data.id}`,
                flashMessages: state.flashMessages.concat([{'severity': 'notice', 'text': data.message}])
              }))
              history.replaceState(
                {},
                document.title,
                `${window.location.origin}/projects/${this.props.projectId}/batches/${data.id}/edit`
              )
            } else {
              this.setState(state => ({
                flashMessages: state.flashMessages.concat([{'severity': 'notice', 'text': data.message}])
              }))
            }
          })
          this.enable(['batch[submit]', 'batch[submit_and_close]'])
        }
      } else {
        response.json().then(data => {
          this.setState(state => ({
            errors       : data.errors,
            flashMessages: state.flashMessages.concat([{'severity': 'error', 'text': data.message}])
          }))
          this.enable(['batch[submit]', 'batch[submit_and_close]'])
        })
      }
    })
  }

  render() {
    return (
      <React.Fragment>
        <Layout
          className="batch-form"
          title={this.state.formAction === 'POST' ? 'Create a New Batch' : 'Edit Batch'}
          hasBackButton
          subfooter={
            <React.Fragment>
              <Button
                value={this.state.formAction === 'POST' ? 'create & close' : 'save & close'}
                type="submit"
                form="batch-form--form"
                disabled={this.state.disabled['batch[submit_and_close]']}
                onClick={() => this.setState({closeOnSuccess: true})}
              />
              <ButtonBorder
                value={this.state.formAction === 'POST' ? 'create' : 'save'}
                type="submit"
                form="batch-form--form"
                disabled={this.state.disabled['batch[submit]']}
                onClick={() => this.setState({closeOnSuccess: false})}
              />
              <ButtonBorder
                value="cancel"
                onClick={this.goBack}
              />
            </React.Fragment>
          }
        >
          <Flash messages={this.state.flashMessages} />

          <form className="batch-form--form" id="batch-form--form" onSubmit={this.handleSubmit}>
            <InputFieldDropdown
              title="specialty"
              required
              name="batch[specialty_id]"
              value={this.state['batch[specialty_id]']}
              errorText={this.state.errors['batch[specialtyId]']}
              eventHandler={this.handleFieldChange}
              selectEntries={Object.entries(this.props.specialties)}
              symbolLimit={0}
            />

            <InputFieldWhiteDark
              title="name"
              required
              name="batch[name]"
              value={this.state['batch[name]']}
              errorText={this.state.errors['batch[name]']}
              eventHandler={this.handleFieldChange}
              placeholder="Provide a clear title for the batch"
              symbolLimit={100}
            />

            <InputFieldDescriptionMiddle
              title="the goal of the batch"
              required
              name="batch[goal]"
              value={this.state['batch[goal]']}
              errorText={this.state.errors['batch[goal]']}
              eventHandler={this.handleFieldChange}
              placeholder="Let people know what will be the result of this batch being completed (ex: When this batch is finished, the xxxx will now do yyyy)"
              symbolLimit={250}
            />

            <InputFieldDescription
              title="description"
              required
              name="batch[description]"
              value={this.state['batch[description]']}
              errorText={this.state.errors['batch[description]']}
              eventHandler={this.handleFieldChange}
              placeholder="Provide a longer description about the batch focus, the type of work involved, and how it will relate to the larger project"
              symbolLimit={750}
            />

            <InputFieldUploadFile
              title="diagram or screenshot"
              name="batch[diagram]"
              errorText={this.state.errors['batch[diagram]']}
              imgPreviewUrl={this.props.batch.diagramUrl}
              imgPreviewDimensions="100x100"
            />

            <input
              type="hidden"
              name="authenticity_token"
              value={this.props.csrfToken}
              readOnly
            />
          </form>
        </Layout>
      </React.Fragment>
    )
  }
}

BatchForm.propTypes = {
  batch       : PropTypes.object.isRequired,
  specialties : PropTypes.object.isRequired,
  formUrl     : PropTypes.string.isRequired,
  formAction  : PropTypes.string.isRequired,
  urlOnSuccess: PropTypes.string.isRequired,
  projectId   : PropTypes.number.isRequired,
  csrfToken   : PropTypes.string.isRequired
}
BatchForm.defaultProps = {
  batch       : {'default': '_'},
  specialties : {'default': '_'},
  formUrl     : '/',
  formAction  : 'POST',
  urlOnSuccess: '/',
  projectId   : 0,
  csrfToken   : '00'
}
export default BatchForm
