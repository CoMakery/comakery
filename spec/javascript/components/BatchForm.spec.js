import React from 'react'
import { mount } from 'enzyme'
import BatchForm from 'components/BatchForm'

describe('BatchForm', () => {
  it('renders correctly without props', () => {
    const wrapper = mount(<BatchForm />)

    expect(wrapper).toMatchSnapshot()

    expect(wrapper.exists('Layout[navTitle]')).toBe(true)
    expect(wrapper.exists('.batch-form')).toBe(true)
    expect(wrapper.exists('ButtonBorder[value="cancel"]')).toBe(true)
    expect(wrapper.exists('ButtonBorder[value="create"]')).toBe(true)
    expect(wrapper.exists('Button[value="create & close"]')).toBe(true)
    expect(wrapper.exists('.batch-form--message')).toBe(false)
    expect(wrapper.exists('.batch-form--form')).toBe(true)
    expect(wrapper.exists('#batch-form--form')).toBe(true)

    expect(wrapper.exists(
      'InputFieldDropdown[title="specialty"][required][name="batch[specialty_id]"]'
    )).toBe(true)

    expect(wrapper.exists(
      'InputFieldWhiteDark[title="name"][required][name="batch[name]"]'
    )).toBe(true)

    expect(wrapper.exists(
      'InputFieldDescriptionMiddle[title="the goal of the batch"][required][name="batch[goal]"]'
    )).toBe(true)

    expect(wrapper.exists(
      'InputFieldDescription[title="description"][required][name="batch[description]"]'
    )).toBe(true)

    expect(wrapper.exists(
      'InputFieldUploadFile[title="diagram or screenshot"][name="batch[diagram]"]'
    )).toBe(true)

    expect(wrapper.exists(
      'input[type="hidden"][name="authenticity_token"]'
    )).toBe(true)
  })

  it('renders correctly with specialties', () => {
    const specialties = {
      '1': 1,
      '2': 2
    }
    const wrapper = mount(<BatchForm specialties={specialties} />)

    expect(wrapper.find(
      'InputFieldDropdown[title="specialty"][required][name="batch[specialty_id]"]'
    ).props().value).toBe(1)

    expect(wrapper.find(
      'InputFieldDropdown[title="specialty"][required][name="batch[specialty_id]"]'
    ).props().selectEntries).toEqual(Object.entries(specialties))
  })

  it('renders correctly with a batch', () => {
    const batch = {
      'id'         : 10,
      'name'       : 'Batch name',
      'description': 'Batch description',
      'goal'       : 'Batch goal',
      'specialtyId': 'writing'
    }
    const specialties = {
      'audio or Video Production'    : 'audio_video_production',
      'community Development'        : 'community_development',
      'data Gathering'               : 'data_gathering',
      'marketing \u0026 Social Media': 'marketing_social',
      'software Development'         : 'software_development',
      'uX :: UI Design'              : 'design',
      'writing'                      : 'writing'
    }

    const wrapper = mount(<BatchForm
      batch={batch}
      specialties={specialties}
    />)

    expect(wrapper.find(
      'InputFieldDropdown[title="specialty"][required][name="batch[specialty_id]"]'
    ).props().value).toBe('writing')

    expect(wrapper.find(
      'InputFieldWhiteDark[title="name"][required][name="batch[name]"]'
    ).props().value).toBe('Batch name')

    expect(wrapper.find(
      'InputFieldDescriptionMiddle[title="the goal of the batch"][required][name="batch[goal]"]'
    ).props().value).toBe('Batch goal')

    expect(wrapper.find(
      'InputFieldDescription[title="description"][required][name="batch[description]"]'
    ).props().value).toBe('Batch description')
  })

  it('renders correctly with csrfToken', () => {
    const wrapper = mount(<BatchForm csrfToken="test" />)

    expect(wrapper.find(
      'input[type="hidden"][name="authenticity_token"]'
    ).props().value).toBe('test')
  })

  it('uses formUrl', () => {
    const wrapper = mount(<BatchForm formUrl="/test" />)

    expect(wrapper.state('formUrl')).toBe('/test')
  })

  it('uses formAction', () => {
    const wrapper = mount(<BatchForm formAction="PUT" />)

    expect(wrapper.state('formAction')).toBe('PUT')
  })

  it('uses urlOnSuccess', () => {
    const wrapper = mount(<BatchForm urlOnSuccess="/test" />)

    expect(wrapper.props().urlOnSuccess).toBe('/test')
  })

  it('displays messages', () => {
    const wrapper = mount(<BatchForm />)

    wrapper.setState({
      flashMessages: [
        {
          'severity': 'notice',
          'text'    : 'notice text'
        },
        {
          'severity': 'warning',
          'text'    : 'warning text'
        },
        {
          'severity': 'error',
          'text'    : 'error text'
        },
      ]
    })

    wrapper.update()

    expect(wrapper.exists('Flash')).toBe(true)
  })

  it('displays errors', () => {
    const wrapper = mount(<BatchForm />)

    wrapper.setState({
      errors: {
        'batch[name]'       : 'name error',
        'batch[description]': 'description error',
      }
    })

    wrapper.update()

    expect(wrapper.exists(
      'InputFieldWhiteDark[errorText="name error"][title="name"][required][name="batch[name]"]'
    )).toBe(true)

    expect(wrapper.exists(
      'InputFieldDescription[errorText="description error"][title="description"][required][name="batch[description]"]'
    )).toBe(true)
  })
})
