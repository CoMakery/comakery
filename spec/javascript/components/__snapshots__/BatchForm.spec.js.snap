// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatchForm renders correctly without props 1`] = `
<BatchForm
  batch={
    Object {
      "default": "_",
    }
  }
  csrfToken="00"
  formAction="POST"
  formUrl="/"
  projectId={0}
  specialties={
    Object {
      "default": "_",
    }
  }
  urlOnSuccess="/"
>
  <Layout
    category=""
    className="batch-form"
    hasBackButton={true}
    navTitle={null}
    sidebar={null}
    subfooter={
      <React.Fragment>
        <Button
          className=""
          disabled={false}
          form="batch-form--form"
          onClick={[Function]}
          type="submit"
          value="create & close"
        />
        <ButtonBorder
          className=""
          form="batch-form--form"
          onClick={[Function]}
          type="submit"
          value="create"
        />
        <ButtonBorder
          className=""
          onClick={[Function]}
          value="cancel"
        />
      </React.Fragment>
    }
    title="Create a New Batch"
  >
    <div
      className="layout batch-form"
    >
      <div
        className="layout--back-button"
      >
        <Icon
          className=""
          name="iconCloseCopy.svg"
          onClick={[Function]}
        >
          <img
            className="icon icon__iconCloseCopy-svg"
            onClick={[Function]}
            src="test-file-stub"
          />
        </Icon>
      </div>
      <div
        className="layout--content"
      >
        <div
          className="layout--content--title"
        >
          Create a New Batch
        </div>
        <hr
          className="layout--content--hr"
        />
        <div
          className="layout--content--wrapper"
        >
          <div
            className="layout--content--content"
          >
            <Flash
              className=""
              messages={Array []}
            >
              <div
                className="flash-message-container"
              />
            </Flash>
            <form
              className="batch-form--form"
              id="batch-form--form"
              onSubmit={[Function]}
            >
              <InputFieldDropdown
                eventHandler={[Function]}
                name="batch[specialty_id]"
                required={true}
                selectEntries={
                  Array [
                    Array [
                      "default",
                      "_",
                    ],
                  ]
                }
                symbolLimit={0}
                title="specialty"
                value="_"
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__dropdown"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="40x40"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  max=""
                  min=""
                  name="batch[specialty_id]"
                  pattern=".*"
                  placeholder="Please enter value"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={
                    Array [
                      Array [
                        "default",
                        "_",
                      ],
                    ]
                  }
                  step=""
                  symbolLimit={0}
                  title="specialty"
                  type="select"
                  value="_"
                >
                  <div
                    className="input-field input-field__dropdown"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        specialty
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <select
                        className="input-field--content__select"
                        disabled={false}
                        name="batch[specialty_id]"
                        onChange={[Function]}
                        required={true}
                        value="_"
                      >
                        <option
                          key="_"
                          value="_"
                        >
                          default
                        </option>
                      </select>
                    </div>
                  </div>
                </InputField>
              </InputFieldDropdown>
              <InputFieldWhiteDark
                eventHandler={[Function]}
                name="batch[name]"
                placeholder="Provide a clear title for the batch"
                required={true}
                symbolLimit={100}
                title="name"
                value=""
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__white__dark"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="40x40"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  max=""
                  min=""
                  name="batch[name]"
                  pattern=".*"
                  placeholder="Provide a clear title for the batch"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  step=""
                  symbolLimit={100}
                  title="name"
                  type="text"
                  value=""
                >
                  <div
                    className="input-field input-field__white__dark"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        name
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                      <span
                        className="input-field--title--counter"
                      >
                        0
                        /
                        100
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <input
                        className="input-field--content__text"
                        name="batch[name]"
                        onChange={[Function]}
                        onClick={null}
                        pattern=".*"
                        placeholder="Provide a clear title for the batch"
                        readOnly={false}
                        required={true}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </InputField>
              </InputFieldWhiteDark>
              <InputFieldDescriptionMiddle
                eventHandler={[Function]}
                name="batch[goal]"
                placeholder="Let people know what will be the result of this batch being completed (ex: When this batch is finished, the xxxx will now do yyyy)"
                required={true}
                symbolLimit={250}
                title="the goal of the batch"
                value=""
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__description-middle"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="40x40"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  max=""
                  min=""
                  name="batch[goal]"
                  pattern=".*"
                  placeholder="Let people know what will be the result of this batch being completed (ex: When this batch is finished, the xxxx will now do yyyy)"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  step=""
                  symbolLimit={250}
                  title="the goal of the batch"
                  type="textarea"
                  value=""
                >
                  <div
                    className="input-field input-field__description-middle"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        the goal of the batch
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                      <span
                        className="input-field--title--counter"
                      >
                        0
                        /
                        250
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <textarea
                        className="input-field--content__text"
                        name="batch[goal]"
                        onChange={[Function]}
                        pattern=".*"
                        placeholder="Let people know what will be the result of this batch being completed (ex: When this batch is finished, the xxxx will now do yyyy)"
                        readOnly={false}
                        required={true}
                        value=""
                      />
                    </div>
                  </div>
                </InputField>
              </InputFieldDescriptionMiddle>
              <InputFieldDescription
                eventHandler={[Function]}
                name="batch[description]"
                placeholder="Provide a longer description about the batch focus, the type of work involved, and how it will relate to the larger project"
                required={true}
                symbolLimit={750}
                title="description"
                value=""
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__description"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="40x40"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  max=""
                  min=""
                  name="batch[description]"
                  pattern=".*"
                  placeholder="Provide a longer description about the batch focus, the type of work involved, and how it will relate to the larger project"
                  readOnly={false}
                  recommended={false}
                  required={true}
                  selectEntries={Array []}
                  step=""
                  symbolLimit={750}
                  title="description"
                  type="textarea"
                  value=""
                >
                  <div
                    className="input-field input-field__description"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        description
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        required
                      </span>
                      <span
                        className="input-field--title--counter"
                      >
                        0
                        /
                        750
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <textarea
                        className="input-field--content__text"
                        name="batch[description]"
                        onChange={[Function]}
                        pattern=".*"
                        placeholder="Provide a longer description about the batch focus, the type of work involved, and how it will relate to the larger project"
                        readOnly={false}
                        required={true}
                        value=""
                      />
                    </div>
                  </div>
                </InputField>
              </InputFieldDescription>
              <InputFieldUploadFile
                imgPreviewDimensions="100x100"
                name="batch[diagram]"
                title="diagram or screenshot"
              >
                <InputField
                  checkboxText=""
                  checked={false}
                  className="input-field__upload-file"
                  copyOnClick={false}
                  disabled={false}
                  errorText=""
                  eventHandler={[Function]}
                  imgInputRef={null}
                  imgPreviewDimensions="100x100"
                  imgPreviewUrl=""
                  imgRequirements=""
                  imgVerifier={[Function]}
                  max=""
                  min=""
                  name="batch[diagram]"
                  pattern=".*"
                  placeholder="Please enter value"
                  readOnly={false}
                  recommended={false}
                  required={false}
                  selectEntries={Array []}
                  step=""
                  symbolLimit={0}
                  title="diagram or screenshot"
                  type="file"
                  value=""
                >
                  <div
                    className="input-field input-field__upload-file"
                  >
                    <div
                      className="input-field--title"
                    >
                      <span
                        className="input-field--title--title"
                      >
                        diagram or screenshot
                      </span>
                      <span
                        className="input-field--title--required"
                      >
                        optional
                      </span>
                    </div>
                    <div
                      className="input-field--content"
                    >
                      <input
                        accept="image/*"
                        className="input-field--content__file"
                        name="batch[diagram]"
                        onChange={[Function]}
                        required={false}
                        type="file"
                      />
                      <span
                        className="input-field--content__file--requirements"
                      />
                    </div>
                  </div>
                </InputField>
              </InputFieldUploadFile>
              <input
                name="authenticity_token"
                readOnly={true}
                type="hidden"
                value="00"
              />
            </form>
          </div>
        </div>
      </div>
      <div
        className="layout--subfooter"
      >
        <div
          className="layout--subfooter--buttons"
        >
          <Button
            className=""
            disabled={false}
            form="batch-form--form"
            onClick={[Function]}
            type="submit"
            value="create & close"
          >
            <input
              className="button"
              disabled={false}
              form="batch-form--form"
              onClick={[Function]}
              type="submit"
              value="create & close"
            />
          </Button>
          <ButtonBorder
            className=""
            form="batch-form--form"
            onClick={[Function]}
            type="submit"
            value="create"
          >
            <Button
              className="button__border"
              disabled={false}
              form="batch-form--form"
              onClick={[Function]}
              type="submit"
              value="create"
            >
              <input
                className="button button__border"
                disabled={false}
                form="batch-form--form"
                onClick={[Function]}
                type="submit"
                value="create"
              />
            </Button>
          </ButtonBorder>
          <ButtonBorder
            className=""
            onClick={[Function]}
            value="cancel"
          >
            <Button
              className="button__border"
              disabled={false}
              onClick={[Function]}
              type="button"
              value="cancel"
            >
              <input
                className="button button__border"
                disabled={false}
                onClick={[Function]}
                type="button"
                value="cancel"
              />
            </Button>
          </ButtonBorder>
        </div>
      </div>
    </div>
  </Layout>
</BatchForm>
`;
